### DISCUSSION TITLE
"curl: (52) Empty reply from server" after start proxy.

### DISCUSSION THREAD

**Discussion by korsunav** [v2rayA/v2rayA#1471] [General]
I run v2rayA on Oracle Linux 9 by docker compose with this settings:
```services:
  v2raya:
    restart: always
    privileged: true
    network_mode: host
    container_name: v2raya
    environment:
      - V2RAYA_V2RAY_BIN=/usr/local/bin/v2ray
      - V2RAYA_LOG_FILE=/tmp/v2raya.log
      - V2RAYA_NFTABLES_SUPPORT=off
      - IPTABLES_MODE=legacy
    volumes:
      - '/etc/v2raya:/etc/v2raya'
      - '/etc/resolv.conf:/etc/resolv.conf'
      - '/lib/modules:/lib/modules:ro'
    image: 'mzz2017/v2raya:latest'
```
Also try with Oracle Linux 8

selinux and firewalld is disabled

If the "Transparent Proxy/System Proxy" set to "Off", then my curl requests going ok
but if i set "Transparent Proxy/System Proxy" to "On. Do not Split Traffic" my curl requests return: 
```curl: (52) Empty reply from server```

My proxy is runing by 3X-UI (v.2.3.12) and it perfect work with Windows, MacOS, iOS and Android clients but not with v2rayA on my Linux machine

v2raya settings:
```
Transparent Proxy/System Proxy - On. Do not Split Traffic
IP Forward - on
Transparent Proxy/System Proxy Implementation - redirect
Traffic Splitting Mode of Rule Port - RoutingA #just 1 rule "default: proxy"
```
Other options is by default

Is any idea what i am doing wrong?

**Comment by eronez** [comment]
What's your curl request and any log info?
Do you have tried any other url or other transparent proxy mode?

**Reply by korsunav** [reply]
Thanx for reply
My curl request is ```curl ifconfig.io``` to see which my public ip address for now
Also i try curl requests to different resources like google, github and etc

And i try all other transparent proxy mode that are in v2raya

Output curl request with --verbose option and proxy **runing**:
```
[root@localhost ~]# curl ifconfig.io -v
*   Trying 188.114.96.3:80...
* Connected to ifconfig.io (188.114.96.3) port 80 (#0)
> GET / HTTP/1.1
> Host: ifconfig.io
> User-Agent: curl/7.76.1
> Accept: */*
>
* Empty reply from server
* Closing connection 0
curl: (52) Empty reply from server
```
Output curl request with --verbose option and proxy **stoped**:
```
[root@localhost ~]# curl ifconfig.io -v
*   Trying 188.114.96.3:80...
* Connected to ifconfig.io (188.114.96.3) port 80 (#0)
> GET / HTTP/1.1
> Host: ifconfig.io
> User-Agent: curl/7.76.1
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Date: Sun, 11 Aug 2024 10:47:28 GMT
< Content-Type: text/plain; charset=utf-8
< Content-Length: 15
< Connection: keep-alive
< CF-Cache-Status: DYNAMIC
< Report-To: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v4?s=Wa7ODFKy23ru8dGk51bbzP0N4R%2F3tL6fV93quYz20przfdU00PGSbxEtVku%2B6ZH5ypV%2F9zSaYZpNtZ0gFpe11oyuoOQ4PjLoWO9XhIBW591Uh7mcalZYA2iZBC2SBQ%3D%3D"}],"group":"cf-nel","max_age":604800}
< NEL: {"success_fraction":0,"report_to":"cf-nel","max_age":604800}
< Server: cloudflare
< CF-RAY: 8b17a5d19b640a21-ARN
< alt-svc: h3=":443"; ma=86400
<
<my_public_ip_adress>
```

**Reply by korsunav** [reply]
This is from v2raya.log after start runing proxy:
```
2024/08/11 10:56:20.736  [I] [io.go:431]  V2Ray 5.16.1 (V2Fly, a community-driven edition of V2Ray.) Custom (go1.22.2 linux/amd64)
2024/08/11 10:56:20.736  [I] [io.go:431]  A unified platform for anti-censorship.
2024/08/11 10:56:21.279  [I] [io.go:431]  127.0.0.1:35638 accepted tcp:127.0.0.1:0 [api-out]
```
This is from v2raya.log after curl request:
```
2024/08/11 10:59:48.053  [I] [io.go:431]  192.168.88.112:60706 accepted tcp:188.114.97.3:80 [proxy]
```

**Reply by eronez** [reply]
@korsunav
On my machine, everything works fine.(redirect tproxy systemtun gvisortun)
Maybe you should check your proxy server, DNS settings of v2raya.
Or try xray core in v2raya(due to the 3X-UI)

**Reply by korsunav** [reply]
Is it possible to change core from v2ray-core to Xray-core in docker? Or i need to build my own image with Xray-core?

**Reply by eronez** [reply]
@korsunav 
https://v2raya.org/en/docs/prologue/installation/docker/

> the default core of v2rayA is xray.

**Reply by korsunav** [reply]
@eronez Thank you very much!
I switch to xray-core by changing environment variable:
```
V2RAYA_V2RAY_BIN=/usr/local/bin/v2ray
```
to
```
V2RAYA_V2RAY_BIN=/usr/local/bin/xray
```
Now it working! Thanx again!
